Sources/Boutique/Store.swift

```diff
@@ -126,12 +108,9 @@ public final class Store<Item: Codable & Equatable>: ObservableObject {
     /// avoid making multiple separate dispatches to the `@MainActor`.
     /// - Parameter item: The items you are removing from the `Store`.
     public func remove(_ items: [Item]) async throws {
-        let itemKeys = items.map { $0[keyPath: self.cacheIdentifier] }
-        let cacheKeys = itemKeys.map({ CacheKey($0) })
+        let itemKeys = items.map({ $0[keyPath: self.cacheIdentifier] })
```

Comment:
Make `itemKeys` a Set so that you can get O(1) operation in L117:
`let itemKeys = Set(items.map({ $0[keyPath: self.cacheIdentifier] }))`